### PR TITLE
Fix headless build

### PR DESCRIPTION
**Pull request by SaltyCucumber**
once approved will need to:
- [ ] bump versions to 0.0.7-beta
- [ ] publish all the libs

### CODE REVIEW

File: packages/core/tsconfig.json [hunk 1]
```diff
@@ -2,9 +2,6 @@
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     "rootDir": "./src",
-    "outDir": "./dist",
-    "declaration": true,
```
Comment (hamidra) [inline packages/core/tsconfig.json:6]: Any reason declarations are removed?!

File: packages/injected-wallets/src/injectedWallets.ts [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import type { Injected, InjectedWindow, InjectedAccount } from '@polkadot/extension-inject/types';
-import type { Account, BaseWallet, BaseWalletProvider, WalletMetadata } from '@polkadot-onboard/core';
 import type { Signer } from '@polkadot/types/types';
+import { Account, BaseWallet, BaseWalletProvider, WalletMetadata, WalletType } from '@polkadot-onboard/core';
```
Comment (hamidra) [inline packages/injected-wallets/src/injectedWallets.ts:3]: import as type. We would want the lib to have zero dependencies on polkadot.js for tree shaking. These are not needed at runtime so should be only imported as types.

File: packages/injected-wallets/tsconfig.json [hunk 3]
```diff
@@ -2,9 +2,6 @@
   "extends": "../../tsconfig.base.json",
   "compilerOptions": {
     "rootDir": "./src",
-    "outDir": "./dist",
-    "declaration": true,
```
Comment (hamidra) [inline packages/injected-wallets/tsconfig.json:6]: I believe it would be a good practice to export declarations.

**Review by hamidra — CHANGES_REQUESTED** [review]

File: packages/core/package.json [hunk 4]
```diff
@@ -14,12 +14,12 @@
...
-  "types": "dist/types/index.d.ts",
-  "type": "module",
+  "main": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
   "scripts": {
-    "build": "rm -rf ./dist/ && yarn format && tsc --build",
-    "build:watch": "tsc --watch",
+    "build": "rm -rf dist/ && yarn format && yarn build:esm && yarn build:cjs",
+    "build:esm": "tsc --outDir dist/esm",
+    "build:cjs": "tsc --module CommonJS --outDir dist/cjs",
```
Comment (hamidra) [inline packages/core/package.json:22]: looking at it again I don't think we want to export .cjs here. This lib is only intended to work in browser runtime, so should be imported as esm.

File: packages/core/package.json [hunk 5]
```diff
@@ -14,12 +14,12 @@
   "bugs": "https://github.com/paritytech/polkadot-onboard.git/issues",
   "homepage": "https://github.com/paritytech/polkadot-onboard.git#readme",
   "license": "Apache-2.0",
-  "main": "dist/index.js",
-  "types": "dist/types/index.d.ts",
-  "type": "module",
```
Comment (hamidra) [inline packages/core/package.json:19]: This should bot be removed! This lib is intended to be only imported as esm. This is why type:'module' was added in the new version.

**Review by hamidra — CHANGES_REQUESTED** [review]

**Comment by hamidra** [comment]
Looking to the headless example, The lib seems to be fine. The issue seems to be with the webpack setup. I don't think this changes are needed. 

It should be set to disable the fullySpecified imports:
https://webpack.js.org/configuration/module/#resolvefullyspecified

I'll send a PR to fix the headless example.

**Comment by SaltyCucumber** [comment]
Superseded by https://github.com/paritytech/polkadot-onboard/pull/28

**Pull request closed without merging**